frame/fast-unstake/src/lib.rs

```diff
@@ -401,16 +406,9 @@ pub mod pallet {
 			);
 
 			if unchecked_eras_to_check.is_empty() {
-				// `stash` is not exposed in any era now -- we can let go of them now.
-				let num_slashing_spans = Staking::<T>::slashing_spans(&stash).iter().count() as u32;
-
-				let result = pallet_staking::Pallet::<T>::force_unstake(
-					RawOrigin::Root.into(),
-					stash.clone(),
-					num_slashing_spans,
-				);
+				let result = T::Staking::force_unstake(stash.clone());
 
-				let remaining = T::DepositCurrency::unreserve(&stash, deposit);
+				let remaining = T::Currency::unreserve(&stash, deposit);
 				if !remaining.is_zero() {
 					frame_support::defensive!("`not enough balance to unreserve`");
 					ErasToCheckPerBlock::<T>::put(0);
```

Comment:
You mean `halt`? Yeah, it makes sense, though since it's already implemented - just another conflict to solve.
I like it though, it's much more DRY and clear.

Reply:
I guess I'd prefer this to remain in the scope of your batching PR to avoid confusion and duplication across the two. This can be done when this one is merged while rebasing into batching